xdsl/ir/post_order.py

```diff
@@ -0,0 +1,29 @@
+from collections import deque
+from collections.abc import Iterator
+from dataclasses import dataclass
+
+from xdsl.ir import Block, Operation
+from xdsl.traits import IsTerminator
+
+
+@dataclass(init=False)
+class PostOrderIterator(Iterator[Block]):
+    stack: deque[Block]
+    seen: set[Block]
```

Comment:
Could you add some dostrings, please?

Also, maybe `stack` should be `queue`?

Reply:
Looking back at this I think I implemented it entirely incorrectly. It should iterate through the children of each node before the node itself. I'll fix it and add docstrings